torch/_meta_registrations.py

```diff
@@ -26,9 +26,17 @@
 
 
 aten = torch.ops.aten
+c10d = torch.ops.c10d
 
 _meta_lib_dont_use_me_use_register_meta = torch.library.Library("aten", "IMPL", "Meta")
 
+# HACK - it is wrong to just register c10d ops to AutogradMeta instead of Meta
```

Comment:
Could you explain a bit more? so, allreduce_ should have a derivative, is it just not properly registered?
Since you're creating a different op , isn't it possible to register the derivative, or alternativelly to register it without CompositeExplicit?

Reply:
The existing registration of allreduce_ and other collectives is wrong.  It registers as CompositeExplicit despite not having a derivative formula.

for now i just followed the same pattern.  Registering to AutogradMeta directly here makes things "work" at least well enough for some prototypes, but we should fix it.